### ISSUE TITLE
Please close this issue

### ISSUE THREAD

**Issue by xiangzhang1015** [spolsley/common-spatial-patterns#1]

**Comment by xiangzhang1015** [comment]
And most importantly, for n classes, we call CSP(C1,C2,..,Cn) which returns n filters (e.g., f1, f2, ..., fn). But in real world testing, fo the never seen input sample, we don't which filter we should use.

**Comment by polsley** [comment]
Hi, Xiang,

Sorry I didn't get a chance to respond earlier!  I saw your note asking about the original algorithm and realized that it was a critical oversight that I never linked to the paper which inspired this code.  I studied from the CSP algorithm described in [this paper by Wang et al. in Clinical Neurophysiology](https://www.sciencedirect.com/science/article/pii/S138824579800056X).  That was a relatively early work on the subject, although there's a lot of variants of CSP as you pointed out.

I hope this answers your question and that others will be able to find the original paper a useful reference if they come across similar concerns.  Thanks for asking, and I'm glad to hear you found the code useful!
